src/kafunk/Consumer.fs

```diff
@@ -897,6 +911,70 @@ module Consumer =
           commitCurrentOffsetsOnGroupJoinProc)
         |> Async.Ignore }
 
+
+  /// Starts consumption from the start offset in the given range
+  /// Will stop consuming for each partition once it reaches the max boundary offset
+  /// Will stop and return the messages WITHIN the range if it overconsumes
+  let consumeRange (c:Consumer) (range:OffsetRange) : Async<seq<Message>> = 
+    async { 
+      let getMsgFromRg (msgs : seq<Partition * Message [] * Map<Partition, Offset>>) range =
+        msgs
+          |> Seq.map (fun (p, ms, _) -> p, ms)
+          |> Seq.groupBy (fst)
+          |> Seq.map (fun (p, u) -> 
+                let num = (range.endOS.[p]) - (range.startOS.[p])
+                u
+                |> Seq.map (snd)
+                |> Array.concat
+                //filter them on end offsets
+                |> Seq.take (int32 (num)))
+          |> Seq.concat
+
+      let isSubsetOf (arr1 : 'a []) (arr2 : 'a []) : bool = 
+        let s1 = Set.ofArray arr1
+        let s2 = Set.ofArray arr2
+        s1.IsSubsetOf s2
```

Comment:
Converting arrays to subsets will lose duplicates. This function will return `true` for `isSubsetOf [|1;1|] [|1;2|]` even though neither is a subset of the other.